src/core/thread/types.d

```diff
@@ -0,0 +1,79 @@
+/**
+ * This module provides types and constants used in thread package.
+ *
+ * Copyright: Copyright Sean Kelly 2005 - 2012.
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Sean Kelly, Walter Bright, Alex Rønne Petersen, Martin Nowak
+ * Source:    $(DRUNTIMESRC core/thread/osthread.d)
+ */
+
+module core.thread.types;
+
+/**
+ * Represents the ID of a thread, as returned by $(D Thread.)$(LREF id).
+ * The exact type varies from platform to platform.
+ */
+version (Windows)
+    alias ThreadID = uint;
+else
+version (Posix)
+{
+    import core.sys.posix.pthread;
+
+    alias ThreadID = pthread_t;
+}
+
+struct ll_ThreadData
+{
+    ThreadID tid;
+    version (Windows)
+        void delegate() nothrow cbDllUnload;
+}
+
+version (GNU)
+{
+    import gcc.builtins;
```

Comment:
Import is unused, perhaps you meant to put it in another module you split out?

Reply:
Already putted, but forgot to remove from here.
Fixed.

Reply:
Actually we don't know `GNU_StackGrowsDown` is really defined anywhere or not.

This is why I changed `version = StackGrowsDown` to enum `bool isStackGrowingDown`.

Reply:
> Actually we don't know GNU_StackGrowsDown is really defined anywhere or not.

On the other hand, that’s all the compiler can tell us, so this sentence isn't make sense. :-)

Reply:
> <s>Already putted, but forgot to remove from here.</s>
> 
> Copied and pasted from the original `osthread.d` source. Maybe, in previous, `GNU_StackGrowsDown` was defined inside of `gcc.builtins`.
> 

You can't import version conditions from another module.

Reply:
> Actually we don't know `GNU_StackGrowsDown` is really defined anywhere or not.
> 

Version conditions can only be defined or undefined.